pkg/option/config.go

```diff
@@ -2509,6 +2484,47 @@ func (c *DaemonConfig) populateHostServicesProtos() error {
 	return nil
 }
 
+func (c *DaemonConfig) populateIdentityAllocation() error {
+	c.DisableCiliumEndpointCRD = viper.GetBool(DisableCiliumEndpointCRDName)
+	c.IdentityAllocationMode = viper.GetString(IdentityAllocationMode)
+	c.KVStore = viper.GetString(KVStore)
+
+	if m := viper.GetStringMapString(KVStoreOpt); len(m) != 0 {
+		c.KVStoreOpt = m
+	}
+
+	switch c.IdentityAllocationMode {
+	// This is here for tests. Some call Populate without the normal init
+	case "":
+		c.IdentityAllocationMode = IdentityAllocationModeKVstore
+
+	case IdentityAllocationModeKVstore:
+		// c.IdentityAllocationMode is set above
+
+	case IdentityAllocationModeCRD:
+		if c.KVStore != "" {
+			log.Warningf("Running Cilium in CRD identity allocation mode excludes using kvstore. Unsetting %s", KVStore)
+			c.KVStore = ""
+		}
+
+	default:
+		return fmt.Errorf("Invalid identity allocation mode %q. It must be one of %s or %s", c.IdentityAllocationMode, IdentityAllocationModeKVstore, IdentityAllocationModeCRD)
+	}
+
+	if c.KVStore == "" {
+		if c.IdentityAllocationMode != IdentityAllocationModeCRD {
+			log.Warningf("Running Cilium with %q=%q requires identity allocation via CRDs. Changing %s to %q", KVStore, c.KVStore, IdentityAllocationMode, IdentityAllocationModeCRD)
+			c.IdentityAllocationMode = IdentityAllocationModeCRD
+		}
+		if c.DisableCiliumEndpointCRD {
+			log.Warningf("Running Cilium with %q=%q requires endpoint CRDs. Changing %s to %t", KVStore, c.KVStore, DisableCiliumEndpointCRDName, false)
+			c.DisableCiliumEndpointCRD = false
+		}
```

Comment:
@aanm do you know if this is still true? We have ciliumidentities CRD now, is this left over or we still have a dependency on ciliumendpoint CRD as well?